robocop/checkers/naming.py

```diff
@@ -662,3 +669,59 @@ def find_similar_variables(self, tokens, node):
                     col=token.col_offset,
                 )
             self.variables[normalized_token].add(token.value)
+
+
+class DeprecatedStatementChecker(VisitorChecker):
+    """Checker for finding deprecated statments."""
+
+    reports = (
+        "deprecated-statement",
+    )
+    # deprecated:alternative
+    deprecated_statements_rf3 = {}
+    deprecated_statements_rf4 = {"run keyword unless": "IF/ELSE",
+                                 "run keyword if":     "IF/ELSE"
+                                 }
+    deprecated_statements_rf5 = {"run keyword unless": "IF/ELSE"}
+
+    def visit_SuiteSetup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_TestSetup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_Setup(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_SuiteTeardown(self, node):  # noqa
+        self.check_if_keyword_is_deprecated(node.name, node)
+
+    def visit_TestTeardown(self, node):  # noqa
```

Comment:
One trick I learned recently in Python - if you have the same method body but different method names (typical in visitor pattern) you can do something like:
```
    def visit_SuiteTeardown(self, node):  # noqa
        self.check_if_keyword_is_deprecated(node.name, node)

   visit_TestTeardown = visit_Teardown ... (and other identical methods).. = visit_SuiteTeardown
``

Reply:
was facing issues to apply suggested solution, can post more info later